test/api/test_cp.py

```diff
@@ -46,3 +46,72 @@ def test_failed_api_response(self):
         with patch("requests.get", side_effect=requests.exceptions.RequestException):
             with pytest.raises(SwanRequestError):
                 get_all_cp_machines()
+
+    def test_retrieve_collateral_balance_valid_address(self):
+        # Mock the requests.get method to return a mock response
+        with patch("requests.get") as mock_get:
+            # Set up the mock response
+            mock_response = Mock()
+            mock_response.json.return_value = {
+                "status": "success",
+                "message": "Successfully retrieved collateral balance",
+                "data": {"balance": 100},
+            }
+            mock_get.return_value = mock_response
+
+            # Call the function with a valid computing provider address
+            result = get_collateral_balance("0x1234abcd")
+
+            # Assert that the response is as expected
+            assert result == {
+                "status": "success",
+                "message": "Successfully retrieved collateral balance",
+                "data": {"balance": 100},
+            }
+    # NOTE: do not uncomment before ApiClient changes
```

Comment:
are we planing to chagne the API Client ?

Reply:
No, I need to make changes on TEST by tokens

Reply:
Got it. then lgtm